mlton/atoms/prim-tycons.fun

```diff
@@ -234,4 +234,64 @@ fun layoutApp (c: t,
       else normal ()
    end
 
+fun layoutFormal (c: t, args: (Layout.t * ({isChar: bool}* BindingStrength.t)) vector) =
+   let
+      local
+         open Layout
+         open BindingStrength
+      in
+         val mayAlign = mayAlign
+         val seq = seq
+         val str = str
+      end
+      val ({isChar}, lay) =
+         case Vector.length args of
+            0 => if equals (c, tuple)
+                    then (({isChar = false}, str "unit"))
+                 else
+                    ({isChar = equals (c, defaultChar ())},
+                     (* always use canonical names, even if synonyms exist *)
+                     if equals (c, real RealSize.R32)
```

Comment:
The special cases are necessary because the types might also have other synonyms. benchmark/tests/tensor includes an example, the type real32 may be printed out as elem. I'm not sure if there's a better way to do this though. I only was using those special cases.